HW_4_Task_2/HW_4_Task_2/List.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+
+namespace HW_4_Task_2
+{
+    /// <summary>
+    /// Abstract data type that represents a countable number of ordered values.
+    /// </summary>
+    public class List
+    {
+        private class ListNode
+        {
+            public ListNode NextNode { get; set; }
+
+            public int Value { get; set; }
+
+            public ListNode(int value)
+            {
+                Value = value;
+                NextNode = null;
+            }
+
+            public ListNode(int value, ListNode next)
+            {
+                Value = value;
+                NextNode = next;
+            }
+        }
+
+        private ListNode head;
+        private int listSize = 0;
+
+        private bool CheckIndex(int index)
+            => index < 0 || listSize < index ? false : true;
```

Comment:
```suggestion
            => index >- 0 && listSize >= index;
```
Кажется, после `?` в тернарном операторе вообще никогда не должно быть булевых констант, потому что само условие --- это булево выражение